application/src/main/java/org/opentripplanner/apis/gtfs/mapping/routerequest/ViaLocationMapper.java

```diff
@@ -42,14 +44,25 @@ private static ViaLocation mapViaLocation(Map<String, Map<String, Object>> via)
         (String) visit.get(FIELD_LABEL),
         (Duration) visit.get(FIELD_MINIMUM_WAIT_TIME),
         mapStopLocationIds((List<String>) visit.get(FIELD_STOP_LOCATION_IDS)),
-        List.of()
+        mapCoordinate(visit.get(FIELD_COORDINATE))
       );
     } else {
       throw new IllegalArgumentException("ViaLocation must define either pass-through or visit.");
     }
   }
 
-  private static List<FeedScopedId> mapStopLocationIds(List<String> ids) {
+  private static List<FeedScopedId> mapStopLocationIds(@Nullable List<String> ids) {
+    if (ids == null) {
+      return List.of();
+    }
     return ids.stream().map(FeedScopedId::parse).toList();
   }
+
+  private static List<WgsCoordinate> mapCoordinate(@Nullable Object coordinate) {
+    if (coordinate == null) {
+      return List.of();
+    }
+    var map = (Map<String, Object>) coordinate;
+    return List.of(new WgsCoordinate((Double) map.get("latitude"), (Double) map.get("longitude")));
+  }
```

Comment:
Consider putting the Coordinate mapping into a separate class so it can be reused.

Reply:
The `PlanCoordinateInput` is not defined in this PR. If in use there should already be a mapper for it?

Reply:
Side note, the way how the coordinate is read here is not what I've usually used. Sometimes we have to use this untyped map for parsing the arguments but often it's possible to use the generated types to parse the arguments. @leonardehrenfried originally wrote this class so I'm not sure if we run into some bug or not in this case if we used the generated types.

Reply:
Feel free to change it. I vaguely remember that I tried it but there was a problem. Sorry, I cannot remember more.

Reply:
Seems to work https://github.com/opentripplanner/OpenTripPlanner/pull/6490/commits/902a17268ea1949600fdd9ffc93371ff35963b83 . I think the issue might have been for mapping a list of objects. I didn't use the types for the list of via points.

Reply:
The coordinate mapping should be moved out even if there are different ways of doing it. Also, this mapper will then of cause just use the extracted mapping.